reflex/reflex.py

```diff
@@ -349,7 +414,394 @@ def makemigrations(
+                time.sleep(1)
+    console.print("Waiting for the new deployment to come up")
+    # TODO: for overwrite case, poll for the old site to come down
+    backend_up = frontend_up = False
+
+    with console.status("Checking backend ..."):
+        for _ in range(constants.Hosting.BACKEND_POLL_TIMEOUT):
+            if backend_up := hosting.poll_backend(deploy_response.backend_url):
+                break
+            time.sleep(1)
+    if not backend_up:
+        console.print("Backend unreachable")
+    else:
+        console.print("Backend is up")
+
+    with console.status("Checking frontend ..."):
+        for _ in range(constants.Hosting.FRONTEND_POLL_TIMEOUT):
+            if frontend_up := hosting.poll_frontend(deploy_response.frontend_url):
+                break
+            time.sleep(1)
+    if not frontend_up:
+        console.print("frontend is unreachable")
+    else:
+        console.print("frontend is up")
+
+    # TODO: below is a bit hacky, refactor
+    hosting.clean_up()
+    if frontend_up and backend_up:
+        console.print(
+            f"Your site [ {key} ] at {regions} is up: {deploy_response.frontend_url}"
+        )
+        return
+    console.warn(
+        "Your deployment is taking unusually long. Check back later on its status: `reflex deployments status`"
+    )
+
+
+deployments_cli = typer.Typer()
+
+
+@deployments_cli.command(name="list")
+def list_deployments(
+    loglevel: constants.LogLevel = typer.Option(
+        config.loglevel, help="The log level to use."
+    ),
+    as_json: bool = typer.Option(
+        False, "-j", "--json", help="Whether to output the result in json format."
+    ),
+):
+    """List all the hosted instances for the specified app."""
+    console.set_log_level(loglevel)
+    # Check if feature is enabled:
+    if not hosting.feature_enabled():
+        return
+    try:
+        deployments = hosting.list_deployments()
+    except Exception as ex:
+        console.error(f"Unable to list deployments due to: {ex}")
+        raise typer.Exit(1) from ex
+
+    if as_json:
+        console.print(json.dumps(deployments))
+        return
+    try:
+        headers = list(deployments[0].keys())
+        table = [list(deployment.values()) for deployment in deployments]
+        console.print(tabulate(table, headers=headers))
+    except Exception as ex:
+        console.debug(f"Unable to tabulate the deployments due to: {ex}")
+        console.print(str(deployments))
+
+
+@deployments_cli.command(name="delete")
+def delete_deployment(
+    key: str = typer.Argument(..., help="The name of the deployment."),
+    loglevel: constants.LogLevel = typer.Option(
+        config.loglevel, help="The log level to use."
+    ),
+):
+    """Delete a hosted instance."""
+    console.set_log_level(loglevel)
+    # Check if feature is enabled:
+    if not hosting.feature_enabled():
+        return
+    try:
+        hosting.delete_deployment(key)
+    except Exception as ex:
```

Comment:
We could just do the `console.error` in the underlying method so we don't have to catch it twice

Reply:
discussed offline, we'll probably consider this later, keep the current way for now